File: sdk/swift/Sources/HiTechCloudSDK/Support.swift
import Foundation

/// Support: tickets, departments, news, knowledgebase
public class SupportResource {
    private let httpClient: HTTPClient
    public init(httpClient: HTTPClient) { self.httpClient = httpClient }

    /// GET /api/tickets - List support tickets
    public func listTickets(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/tickets", params: params)
    }

    /// POST /api/tickets - Create new support ticket
    public func createTicket(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/tickets", params: params)
    }

    /// GET /api/tickets/{number} - Get ticket details with replies
    public func getTicket(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/tickets/{number}", params: params)
    }

    /// POST /api/tickets/{number} - Reply to ticket
    public func reply(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/tickets/{number}", params: params)
    }

    /// GET /api/ticket/attachment/{file} - Get ticket attachment
    public func getAttachment(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/ticket/attachment/{file}", params: params)
    }

    /// PUT /api/tickets/{number}/open - Re-open closed ticket
    public func reopen(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.put("/api/tickets/{number}/open", params: params)
    }

    /// PUT /api/tickets/{number}/close - Close ticket
    public func close(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.put("/api/tickets/{number}/close", params: params)
    }

    /// GET /api/ticket/departments - List ticket departments
    public func listDepartments(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/ticket/departments", params: params)
    }

    /// GET /api/news - List news
    public func listNews(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/news", params: params)
    }

    /// GET /api/news/{news_id} - Get news item details
    public func getNews(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/news/{news_id}", params: params)
    }

    /// GET /api/knowledgebase - List knowledgebase categories
    public func listKnowledgebaseCategories(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/knowledgebase", params: params)
    }

    /// GET /api/knowledgebase/{category_id} - Get category with articles
    public func getKnowledgebaseCategory(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/knowledgebase/{category_id}", params: params)
    }

    /// GET /api/knowledgebase/article/{article_id} - Get article details
    public func getKnowledgebaseArticle(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/knowledgebase/article/{article_id}", params: params)
    }

}

File: sdk/kotlin/src/main/kotlin/vn/hitechcloud/sdk/Support.kt
package vn.hitechcloud.sdk

/** Support: tickets, departments, news, knowledgebase */
class SupportResource(private val httpClient: HttpClient) {
    /** GET /api/tickets - List support tickets */
    fun listTickets(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/tickets", params)
    }

    /** POST /api/tickets - Create new support ticket */
    fun createTicket(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/tickets", params)
    }

    /** GET /api/tickets/{number} - Get ticket details with replies */
    fun getTicket(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/tickets/{number}", params)
    }

    /** POST /api/tickets/{number} - Reply to ticket */
    fun reply(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/tickets/{number}", params)
    }

    /** GET /api/ticket/attachment/{file} - Get ticket attachment */
    fun getAttachment(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/ticket/attachment/{file}", params)
    }

    /** PUT /api/tickets/{number}/open - Re-open closed ticket */
    fun reopen(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.put("/api/tickets/{number}/open", params)
    }

    /** PUT /api/tickets/{number}/close - Close ticket */
    fun close(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.put("/api/tickets/{number}/close", params)
    }

    /** GET /api/ticket/departments - List ticket departments */
    fun listDepartments(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/ticket/departments", params)
    }

    /** GET /api/news - List news */
    fun listNews(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/news", params)
    }

    /** GET /api/news/{news_id} - Get news item details */
    fun getNews(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/news/{news_id}", params)
    }

    /** GET /api/knowledgebase - List knowledgebase categories */
    fun listKnowledgebaseCategories(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/knowledgebase", params)
    }

    /** GET /api/knowledgebase/{category_id} - Get category with articles */
    fun getKnowledgebaseCategory(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/knowledgebase/{category_id}", params)
    }

    /** GET /api/knowledgebase/article/{article_id} - Get article details */
    fun getKnowledgebaseArticle(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/knowledgebase/article/{article_id}", params)
    }

}
